### PR TITLE
Backport XenopusJamboreeParser from ensembl-xref

### PR DESCRIPTION
**Pull request by mkszuba**
## Description

Port changes made to XenopusJamboreeParser in _ensembl-xref_, including unit tests, back to _ensembl:feature/xref_sprint_.

## Use case

See ENSCORESW-3223. Originally worked on by Prem.

## Benefits

Salvage more work from the 2018 xref sprint.

## Possible Drawbacks

Increased test-suite run time (note: Travis does not automatically execute tests from _misc-scripts/xref_mapping_ yet).

## Testing

_Have you added/modified unit tests to test the changes?_

Yes.

_If so, do the tests pass/fail?_

They pass.

_Have you run the entire test suite and no regression was detected?_

Yes (including tests from _misc-scripts/xref_mapping_), no regression detected.

### CODE REVIEW

**Comment by mkszuba** [comment]
Output comparison with _release/98_ - all differences accounted for:
 * _gene_direct_xref.linkage_xref_ is now NULL where it was an empty string with the old version of the parser,
 * the ", X of Y" postfix has been removed from those (very) few _xref.description_ values which used to feature it.

File: misc-scripts/xref_mapping/t/test-data/xenopusjamboree.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+XB-GENE-478054	trnt1	tRNA nucleotidyl transferase, CCA-adding, 1	ENSXETG00000025091
+XB-GENE-478064	foxh1.2	forkhead box H1, gene 2	ENSXETG00000005286
+XB-GENE-478074	nr5a2	nuclear receptor subfamily 5 group A member 2	ENSXETG00000000314
+XB-GENE-478084	tbx1	T-box 1	ENSXETG00000006304
+XB-GENE-478094	nr1d1	nuclear receptor subfamily 1 group D member 1	ENSXETG00000024397
+XB-GENE-478104	nucb1	nucleobindin 1	ENSXETG00000021229
+XB-GENE-478113	nsa2	NSA2, ribosome biogenesis homolog	ENSXETG00000005077
+XB-GENE-478121	csnk1a1	casein kinase 1 alpha 1	ENSXETG00000020861
+XB-GENE-478131	hoxc6	homeobox C6	ENSXETG00000023479
+XB-GENE-478141	hba1	hemoglobin subunit alpha 1	ENSXETG00000025664
```
Comment (ens-bwalts) [inline misc-scripts/xref_mapping/t/test-data/xenopusjamboree.txt]: Two lines to be added from a recent download that can test the two substitution regexes in `parse_description`

`XB-GENE-940866  rtp3c   receptor (chemosensory) transporter protein 3 gene C [provisional]      ENSXETG00000019753`
`XB-GENE-981482  or1e2l  conserved hypothetical olfactory receptor, 8 of 17      ENSXETG00000026609`

File: misc-scripts/xref_mapping/t/xenopus.t [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  {
+   accession   => "XB-GENE-478054",
+   label       => 'trnt1',
+   description => 'tRNA nucleotidyl transferase, CCA-adding, 1',
+   source_id   => 150,
+   species_id  => 8364
+  }
+ ),
+ 'Sample frog direct Xref has been inserted'
+);
```
Comment (ens-bwalts) [inline misc-scripts/xref_mapping/t/xenopus.t]: Check that the two lines that test the regexes in `parse_description` get added correctly

**Comment by mkszuba** [comment]
Done and done!

**Pull request merged**